rotary_bed_MO.py

```diff
@@ -0,0 +1,252 @@
+import numpy as np
+import pandas as pd
+import pyomo.environ as pyo
+from measure_optimize import (
+    MeasurementOptimizer,
+    SensitivityData,
+    MeasurementData,
+    CovarianceStructure,
+    ObjectiveLib,
+)
+import pickle
+import time
+
+
+# set up problem formulation
+
+# number of time points for DCM
+Nt = 110
+# maximum manual measurement number for each measurement
+max_manual_num = 5
+# minimal measurement interval
+min_interval_num = 10.0
+# maximum manual measurement number for all measurements
+total_max_manual_num = 20
+# index of columns of SCM and DCM in Q
+static_ind = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+dynamic_ind = [11, 12, 13, 14, 15]
+# this index is the number of SCM + nubmer of DCM, not number of DCM timepoints
+all_ind = static_ind + dynamic_ind
+num_total_measure = len(all_ind)
+# meausrement names
+all_names_strategy3 = [
+    "Ads.gas_inlet.F",
+    "Ads.gas_outlet.F",
+    "Ads.gas_outlet.T",
+    "Des.gas_inlet.F",
+    "Des.gas_outlet.F",
+    "Des.gas_outlet.T",
+    "Ads.T_g.Value(19,10)",
+    "Ads.T_g.Value(23,10)",
+    "Ads.T_g.Value(28,10)",  # all static
+    'Ads.gas_outlet.z("CO2").static',
+    'Des.gas_outlet.z("CO2").static',  # static z
+    'Ads.gas_outlet.z("CO2").dynamic',
+    'Des.gas_outlet.z("CO2").dynamic',  # dynamic z
+    'Ads.z("CO2",19,10)',
+    'Ads.z("CO2",23,10)',
+    'Ads.z("CO2",28,10)',
+]
+# define error variance
+error_variance = [1, 1, 1, 1, 1, 1, 1, 1, 1, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
+# define error matrix
+error_mat = [[0] * len(all_names_strategy3) for _ in range(len(all_names_strategy3))]
+
+# set up variance in the diagonal elements
+for _ in range(len(all_names_strategy3)):
```

Comment:
Use i or something else instead of _ as an index. That is more common.

Reply:
done